README.md

```diff
@@ -0,0 +1,284 @@
+# nurbs
+
+> [Non-Uniform Rational B-Splines][nurbs] (NURBS) of any dimensionality
+
+This library implements n-dimensional [Non-Uniform Rational B-Splines][nurbs] (NURBS). It has no dependencies and uses code generation to unroll loops, optimize for various cases (uniform and non-uniform; rational and non-rational; clamped, open, and periodic) and allow compatibility with multiple input types (arrays of arrays, [ndarrays][ndarray]).
+
+## Installation
+
+**Not yet published.**
+
+```bash
+$ npm install nurbs
+```
+
+## Examples
+
+### Open B-Spline
+
+To construct an open quadratic B-Spline in two dimensions:
+<!--The following example, instead of creating a new spline, this example updates the existing spline by calling the existing spline as a constructor, which re-runs the validations while preserving the same instance. (This feature may be removed. It's simple but weird and since the splines have very little internal state, it really doens't save much speed or allocation.) In general, references to the input data are retained so that you may mutate the data (including the number of points) after instantiation. If you change the degree or dimensionality of the curve though, you must either create a new curve or invoke the existing curve as a constructor and pass new data.-->
+
+```javascript
+var nurbs = require('nurbs');
+
+var curve = nurbs({
+  points: [[-1, 0], [-0.5, 0.5], [0.5, -0.5], [1, 0]],
+  degree: 2
+});
+
+curve.domain;
+// => [[2, 4]]
+
+curve.evaluate([], 3.0);
+// => [0, 0]
+```
+
+If you don't provide a knot vector, a uniform knot vector with integer values will be provided implicitly. In this case the knots would be `[0, 1, 2, 3, 4, 5, 6]`. The above example queries the valid domain using `curve.domain`, which returns a nested array since each dimension is parameterized separately. In this case that's just one dimension, from `t = 2` to `t = 4`. Plotting shows the open (unclamped) spline:
+
+<p align="center">
+<img width="480" src="./docs/open.png" alt="Open B-Spline">
+</p>
+
+### Clamped B-Spline
+
+To construct a clamped spline, that is, a spline which passes through its endpoints, you may specify boundary conditions with the `boundary` property:
+
+```javascript
+curve = nurbs({
```

Comment:
Sometimes people are ignorant and don't understand that examples lead off of one another. It might be good to add the `require` and `var` to this one too

Reply:
Alright to add a `var` but not a `require`? There's only one require identically named as the module, so I left that part sorta implicit.